src/client.ts

```diff
@@ -43,22 +44,37 @@ const arrow = grpc.loadPackageDefinition(packageDefinition).arrow as any;
 const flight_proto = arrow.flight.protocol;
 
 class SpiceClient {
-  private _apiKey: string;
+  private _apiKey?: string;
   private _flight_url: string;
   private _http_url: string;
+  private _flight_ssl_enabled: boolean = true;
   private _maxRetries: number = retry.FLIGHT_QUERY_MAX_RETRIES;
 
-  public constructor(
-    apiKey: string,
-    http_url: string = 'https://data.spiceai.io',
-    flight_url: string = 'flight.spiceai.io:443'
-  ) {
-    this._apiKey = apiKey;
-    this._http_url = http_url;
-    this._flight_url = flight_url;
+  public constructor({
```

Comment:
We can do the following to not break the most common use-case where API is provided as param. 
```
  public constructor(config: SpiceClientConfig | string ) {
    if (typeof config === 'string') {
      config = { api_key: params };
    }
```

Reply:
Thanks @sgrebnov! Updated